MigrationTutorial/Services/RealmService.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.IO;
+using Realms;
+using MigrationTutorial.Migrations;
+using MigrationTutorial.Utils;
+using Realms.Schema;
+
+namespace MigrationTutorial.Services
+{
+    public class RealmService
+    {
+        private static ulong _schemaVersion = 0;
+
+        private static RealmConfiguration _conf;
+
+        public static Realm GetRealm() => Realm.GetInstance(_conf);
+
+        public static void Init(ulong schemaVersion)
+        {
+            if (_schemaVersion == 0)
+            {
+                var realmPath = GetPath();
+                _schemaVersion = schemaVersion;
+
+                if (File.Exists(realmPath) && schemaVersion == 1)
+                {
+                    try
+                    {
+                        Logger.LogDebug($"Since the realm already exists and the supplied schema version is 1, it's assumed you want to start from scratch.\n Deleting {realmPath}");
+                        File.Delete(realmPath);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.LogWarning($"It was not possible to delete the local realm at path {realmPath} because of an exception\n{e}");
+                    }
+                }
+
+                RealmSchema schema = null;
+                if (_schemaVersion < 2)
```

Comment:
No real reason. I just tried to be consistent with that the [docs](https://docs.mongodb.com/realm/sdk/dotnet/examples/modify-an-object-schema/#migration-functions) show.

Reply:
In the migration function it makes sense, because you could need to migrate from version 1 to version 10, and you need to do all the steps of the migration version i to version i+1, without skipping any. 
In your case it's not additive, each version has its own specific schema.